### PR TITLE
Storybook cleanup

### PR DESCRIPTION
**Pull request by skylarmb**
- Update to latest version of storybook
- Fix `yarn.lock` pointing to internal registry
- Update story files to typescript

How to test this branch: 

1. `git checkout sb-storybook-cleanup`
2. `yarn`
3. `yarn storybook`

### CODE REVIEW

File: .storybook/preview.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+
+export const parameters = {
+  actions: { argTypesRegex: "^on[A-Z].*" },
```
Comment (amytang0) [inline .storybook/preview.js:3]: woah so weird. is this basically looking for anything matching `on<X>`?

File: examples/hoc.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      onSuccess={onSuccess}
+      onEvent={onEvent}
+    >
+      Open Link and connect your bank!
+    </PlaidLink>
+  );
+};
+
+// storybook config
+export const HOC = () => <Link token={text('Link token', '')} />;
```
Comment (amytang0) [inline examples/hoc.tsx:39]: what is HOC.

File: package.json [hunk 3]
```diff
@@ -33,9 +33,9 @@
   },
   "scripts": {
     "lint": "eslint '{src,examples}/**/*.{ts,tsx,js,jsx}' --ignore-path .gitignore",
-    "storybook": "start-storybook -p 9001",
+    "storybook": "start-storybook -p 6006",
```
Comment (amytang0) [inline package.json:36]: why change the port?

File: src/types/index.ts [hunk 4]
```diff
@@ -139,7 +139,9 @@ export type PlaidLinkOptionsWithPublicKey = CommonPlaidLinkOptions & {
 export type PlaidLinkOptionsWithLinkToken = CommonPlaidLinkOptions & {
   // Provide a link_token associated with your account. Create one
   // using the /link/token/create endpoint.
-  token: string;
+  // If token is null or undefined, the Link iframe will preloaded, but Link
+  // will not be initialized, and `ready` will be false
+  token?: string;
```
Comment (amytang0) [inline src/types/index.ts:144]: This seems like an easy way our customers can shoot themselves in the foot...   a LinkTokenOptions with no token? I'd rather the beauty of a typing system kick in and say HEY DON'T DO THAT

File: .storybook/preview.js (hunk 1 above)
Comment (skylarmb) [reply .storybook/preview.js:3]: idk this is an storybook autogenerated file 🤔 I think this enables some magic for the new storybook 6 "controls" feature

File: examples/hoc.tsx (hunk 2 above)
Comment (skylarmb) [reply examples/hoc.tsx:39]: [Higher Order Component](https://reactjs.org/docs/higher-order-components.html). Although idk if this even qualifies as one... it has just been the name of this in this repo for a while

File: src/types/index.ts (hunk 4 above)
Comment (skylarmb) [reply src/types/index.ts:144]: oops this was supposed to be in #197 not here... good catch

**Pull request closed without merging**